Skip LoginStatus Table creation in ResourceLoadStatisticsStore.cpp if LoginStatusAPIEnabled is disabled

pooneh-nb opened the pull request:
#### d290aa0222a7da9eea14b04ff7f8cf4078cd68a3
<pre>
Skip LoginStatus Table creation in ResourceLoadStatisticsStore.cpp if LoginStatusAPIEnabled is disabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=279916">https://bugs.webkit.org/show_bug.cgi?id=279916</a>
Radar://136240270

Reviewed by NOBODY (OOPS!).

We should avoid creating LoginStatus Table in ResourceLoadStatisticsStore.cpp if LoginStatusAPIEnabled is disabled.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
(WebKit::ResourceLoadStatisticsStore::addLoginStatusTableIfNecessary):
(WebKit::ResourceLoadStatisticsStore::setLoginStatus):
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.h:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d290aa0222a7da9eea14b04ff7f8cf4078cd68a3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67710 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/47092 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20349 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71764 "Built successfully") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54891 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18657 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54179 "Passed tests") | [⏳ 🧪 wincairo-tests ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70777 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43190 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58558 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34647 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39864 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15964 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/17209 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/60827 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61831 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16306 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73462 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/66957 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11673 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15603 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61631 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11708 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58627 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61678 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9519 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3147 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/88725 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42899 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15648 "Found 4 new JSC stress test failures: wasm.yaml/wasm/function-tests/many-arguments-to-function.js.wasm-slow-memory, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-bbq, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-collect-continuously, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-no-cjit (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43975 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/45162 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43714 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.h:
```diff
@@ -440,6 +442,8 @@ class ResourceLoadStatisticsStore final : public DatabaseUtilities, public CanMa
     bool m_dataRecordsBeingRemoved { false };
     bool m_isNewResourceLoadStatisticsDatabaseFile { false };
+
+    SharedPreferencesForWebProcess m_sharedPreferencesForWebProcess;
```
This isn't what we want to check, this will just construct an empty object (so the feature will always be disabled).

`m_sharedPreferencesForWebProcess` on `NetworkConnectionToWebProcess` is what we need to check.

pooneh-nb committed:
Skip LoginStatus Table creation in ResourceLoadStatisticsStore.cpp if LoginStatusAPIEnabled is disabled
https://bugs.webkit.org/show_bug.cgi?id=279916
Radar://136240270

Reviewed by NOBODY (OOPS!).

We should avoid creating LoginStatus Table in ResourceLoadStatisticsStore.cpp if LoginStatusAPIEnabled is disabled.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
(WebKit::ResourceLoadStatisticsStore::addLoginStatusTableIfNecessary):
(WebKit::ResourceLoadStatisticsStore::setLoginStatus):
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.h:

charliewolfe reviewed (commented):
We still need to remove where the LoginStatus table is currently created

achristensen07 commented on Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.h:
```diff
@@ -27,6 +27,7 @@
 
 #include "DatabaseUtilities.h"
 #include "ResourceLoadStatisticsClassifier.h"
+#include "SharedPreferencesForWebProcess.h"
```
This is no longer needed.

achristensen07 commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -4237,6 +4237,7 @@ LoginStatusAPIEnabled:
       default: false
     WebCore:
       default: false
+  sharedPreferenceForWebProcess: true
```
This is no longer needed.

achristensen07 commented on Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
```diff
@@ -982,6 +982,22 @@ void ResourceLoadStatisticsStore::addMissingTablesIfNecessary()
+{
+    static NeverDestroyed<TableAndIndexPair> missingEntry = std::make_pair<String, std::optional<String>>(createLoginStatus, std::nullopt);
+
+    if (!this->tableExists("LoginStatus"_s)) {
```
LoginStatus should probably also be removed from ResourceLoadStatisticsStore::sortedTables and ResourceLoadStatisticsStore::expectedTableAndIndexQueries
